typespec-extension/src/code-model-builder.ts

```diff
@@ -1302,119 +1305,138 @@ export class CodeModelBuilder {
     });
     op.addParameter(parameter);
 
+    const jsonMergePatch = operationIsJsonMergePatch(sdkHttpOperation);
+
+    const schemaIsPublicBeforeProcess =
+      schema instanceof ObjectSchema && (schema as SchemaUsage).usage?.includes(SchemaContext.Public);
+
     this.trackSchemaUsage(schema, { usage: [SchemaContext.Input] });
 
     if (op.convenienceApi) {
       // model/schema does not need to be Public or Internal, if it is not to be used in convenience API
       this.trackSchemaUsage(schema, { usage: [op.internalApi ? SchemaContext.Internal : SchemaContext.Public] });
     }
 
-    if (operationIsJsonMergePatch(sdkHttpOperation)) {
+    if (jsonMergePatch) {
       this.trackSchemaUsage(schema, { usage: [SchemaContext.JsonMergePatch] });
     }
     if (op.convenienceApi && operationIsMultipart(sdkHttpOperation)) {
       this.trackSchemaUsage(schema, { serializationFormats: [KnownMediaType.Multipart] });
     }
 
-    // Implicit body parameter would have usage flag: UsageFlags.Spread, for this case we need to do body parameter flatten
-    const bodyParameterFlatten = sdkType.kind === "model" && sdkType.usage & UsageFlags.Spread && !this.isArm();
-
-    if (schema instanceof ObjectSchema && bodyParameterFlatten) {
-      // flatten body parameter
-      const parameters = sdkHttpOperation.parameters;
-      const bodyParameter = sdkHttpOperation.bodyParam;
-      // name the schema for documentation
-      schema.language.default.name = pascalCase(op.language.default.name) + "Request";
-
-      if (!parameter.language.default.name) {
-        // name the parameter for documentation
-        parameter.language.default.name = "request";
-      }
+    if (op.convenienceApi) {
+      // Explicit body parameter @body or @bodyRoot would result to the existance of rawHttpOperation.parameters.body.property
+      // Implicit body parameter would result to rawHttpOperation.parameters.body.property be undefined
+      // see https://typespec.io/docs/libraries/http/cheat-sheet#data-types
+      const bodyParameterFlatten =
```

Comment:
Use Chenjie's recommended check for is body spread? https://teams.microsoft.com/l/message/19:meeting_MWQ2ODAwOTgtMmI4Mi00MWY2LTkxZjgtNzE0N2FjOGJhNDVi@thread.v2/1724754309829?context=%7B%22contextType%22%3A%22chat%22%7D

Reply:
Both is temporary solution.

Reply:
May I know why is Chenjie's one temp solution? Is there any futhur plan for checking if a body needs spread? 
And also using Chenjie's check could reduce our usage on raw type.

Reply:
I think I told multiple times, that we should use `SdkServiceMethod` for method signature, not `SdkServiceOperation`. And grouping/flattening is the method signature.

Reply:
I just would like to know do we have furthur plan on how to check if a body needs spread when using `SdkServiceOperation`. 
Yeah, we should adopt `SdkServiceMethod` and I agree if we use `SdkServiceMethod`, we could check grouping/spread there. But I'm not sure when will we adopt `SdkServiceMethod`, if it is not short term, personally I feel it is better to follow the guidance when using `SdkServiceOperation`.

Reply:
We DON'T check body flatten on `SdkServiceOperation`.

I WILL DO THAT WHEN I HAVE TIME, if you don't do that.

Reply:
It's not to check directly on `SdkServiceOperation`, it checks propertiese on  `SdkServiceOperation` like `SdkServiceOperation.bodyParam`. There is limitation if we want to use `SdkServiceMethod`, see this [comment](https://github.com/Azure/autorest.java/pull/2861#issuecomment-2297882441) in my previous pr. That's why I said I don't know if we can use `SdkServiceMethod` soon. (If we can use `SdkMethod`, I will use  it in the begining. It's not about who do it.. I can do it as well when I get time. )
I will check if there is existing TCGC issue on this, previously I just talked with Chenjie offline, and he said getting correct parameters mapping will be supported. If no issue, I will create one.